herd/tests/instructions/AArch64.MTE/L00.litmus

```diff
@@ -0,0 +1,14 @@
+AArch64 L00
+Variant=mixed,mte,sync
+{
+int x[6]={1,2,3,4,5,6};
+0:X1=x;
+0:X2=x:red;
+}
+ P0              ;
+ STZG  X2,[X1]   ;
+ MOV W0,#7       ;
+ STR W0,[X2]     ;
+ STR W0,[X1,#20] ; 
```

Comment:
It's a little weird that we allow an unchecked STR to `x` when previously we had a checked STR `x`. It makes sense to allow unchecked locations and checked locations but mixed uses of the same location is not very realistic. But this might not be in the scope for this PR.

Reply:
Looking at L02 did you mean this to be:
```suggestion
 STR W0,[X2,#20] ; 
```

Reply:
I do not understand your remarks on "unchecked" STR. Which STR are checked or unchecked ?

As to `STR W0,[X1,#20]`, this is what I intended. As I understand it, both `STR` succeeds the first one `STR W0,[X2]` stores into the granule whose color has changed to `:red`, while the second `STR W0,[X1,#20]` sores into the next granule, whose color has not changed and hence stayed `:green`. Assuming a granule size of 16 bytes.

Reply:
Ah sorry, do we assume a default tag (color) for symbolic addresses as well? Meaning x is assumed to be x:green? I was under the impression we only had default values for tag locations (tag(x)).

MTE doesn’t require that all memory instructions perform a tag check. There are cases where the instruction doesn’t perform a tag check (for example stg doesn't check or a load y where y is mapped as Normal Memory and not TaggedNormal doesn’t check either). So I thought, so without checking the code, that if a symbolic address didn’t specify a colour it would signal to herd to slide the tag check.

Reply:
As I understand, all locations have a tag and this tag is green by default. In other words, all locations have the TaggedNormal attribute. Do we understand each other?

Reply:
OK, I think I understand now. In `-variant mte`:
* If tag(x) is not explicitly initialised then,  `herd7` will initialise to `:green"
* If a register is initialised with a symbolic location which doesn't specify a tag (for example `0:X0=x`), `herd7` will assume that a default tag `:green` for this symbolic address (i.e. `0:X0=x:green`).
* All locations are assumed to be mapped as `TaggedNormal`
and now I also understand why `L03` is correctly specified.